zetaclient/bitcoin/bitcoin_client.go

```diff
@@ -721,86 +726,61 @@ func (ob *BTCChainClient) IsInTxRestricted(inTx *BTCInTxEvnet) bool {
 	return false
 }
 
+// GetBtcEvent either returns a valid BTCInTxEvnet or nil
+// Note: the caller should retry the tx on error (e.g., GetSenderAddressByVin failed)
 func GetBtcEvent(
+	rpcClient interfaces.BTCRPCClient,
 	tx btcjson.TxRawResult,
-	targetAddress string,
+	tssAddress string,
 	blockNumber uint64,
-	logger *zerolog.Logger,
+	logger zerolog.Logger,
 	netParams *chaincfg.Params,
 	depositorFee float64,
 ) (*BTCInTxEvnet, error) {
 	found := false
 	var value float64
 	var memo []byte
 	if len(tx.Vout) >= 2 {
-		// first vout must to addressed to the targetAddress with p2wpkh scriptPubKey
-		out := tx.Vout[0]
-		script := out.ScriptPubKey.Hex
-		if len(script) == 44 && script[:4] == "0014" { // segwit output: 0x00 + 20 bytes of pubkey hash
-			hash, err := hex.DecodeString(script[4:])
-			if err != nil {
-				return nil, err
-			}
-			wpkhAddress, err := btcutil.NewAddressWitnessPubKeyHash(hash, netParams)
-			if err != nil {
+		// 1st vout must to addressed to the tssAddress with p2wpkh scriptPubKey
+		vout0 := tx.Vout[0]
+		script := vout0.ScriptPubKey.Hex
+		if len(script) == 44 && script[:4] == "0014" { // P2WPKH output: 0x00 + 20 bytes of pubkey hash
+			receiver, err := DecodeScriptP2WPKH(vout0.ScriptPubKey.Hex, netParams)
+			if err != nil { // should never happen
 				return nil, err
 			}
-			if wpkhAddress.EncodeAddress() != targetAddress {
-				return nil, nil // irrelevant tx to us, skip
+			// skip irrelevant tx to us
+			if receiver != tssAddress {
+				return nil, nil
 			}
 			// deposit amount has to be no less than the minimum depositor fee
-			if out.Value < depositorFee {
-				return nil, fmt.Errorf("btc deposit amount %v in txid %s is less than depositor fee %v", value, tx.Txid, depositorFee)
+			if vout0.Value < depositorFee {
+				logger.Info().Msgf("GetBtcEvent: btc deposit amount %v in txid %s is less than depositor fee %v", vout0.Value, tx.Txid, depositorFee)
+				return nil, nil
```

Comment:
Why is the error return removed here?

Reply:
In the previous code, the caller of this `GetBtcEvent` method will skip the `tx` on error.
This `GetBtcEvent` is now modified to be able to parse different types of sender addresses and RPC query (`GetSenderAddressByVin`) is employed to achieve that goal and `GetBtcEvent` could also fail due to RPC errors.

The current version of `GetBtcEvent` should either 
- return a valid event or nil event (indicating no event found) OR
- OR return an error to indicate it failed to parse the tx due to some reason (mostly RPC error). On error, the caller of `GetBtcEvent` has to re-parse this tx the next time to not miss anything.